src/Stratis.Bitcoin.Tests/Connection/ConnectionManagerControllerTest.cs

```diff
@@ -13,42 +13,39 @@ namespace Stratis.Bitcoin.Tests.Controllers
     public class ConnectionManagerControllerTest : LogsTestBase
     {
         private readonly Mock<IConnectionManager> connectionManager;
-        private ConnectionManagerController controller;
+        private readonly ConnectionManagerController controller;
         private readonly Mock<ILoggerFactory> mockLoggerFactory;
 
         public ConnectionManagerControllerTest()
         {
             this.connectionManager = new Mock<IConnectionManager>();
             this.mockLoggerFactory = new Mock<ILoggerFactory>();
             this.mockLoggerFactory.Setup(i => i.CreateLogger(It.IsAny<string>())).Returns(new Mock<ILogger>().Object);
-            this.connectionManager.Setup(i => i.Network)
-                .Returns(KnownNetworks.StratisTest);
-            this.controller = new ConnectionManagerController(this.connectionManager.Object, this.LoggerFactory.Object);
+            this.connectionManager.Setup(i => i.Network).Returns(this.Network);
+            var peerBanning = new Mock<IPeerBanning>();
+            this.controller = new ConnectionManagerController(this.connectionManager.Object, peerBanning.Object, this.LoggerFactory.Object);
         }
 
         [Fact]
-        public void AddNodeAPI_InvalidCommand_ThrowsArgumentException()
+        public async Task AddNodeAPI_InvalidCommand_ThrowsArgumentExceptionAsync()
         {
             string endpoint = "0.0.0.0";
             string command = "notarealcommand";
 
-            IActionResult result = this.controller.AddNodeAPI(endpoint, command);
+            var result = (JsonResult)await this.controller.AddNodeApiAsync(endpoint, command);
```

Comment:
If an error occurs this will fail with:
```
Message: System.InvalidCastException : Unable to cast object of type 'Stratis.Bitcoin.Utilities.JsonErrors.ErrorResult' to type 'Microsoft.AspNetCore.Mvc.JsonResult'.
```